server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -264,18 +265,13 @@ static void clearLoadedModules()
    */
   public static URLClassLoader getClassLoaderForExtension(File extension) throws MalformedURLException
   {
-    URLClassLoader loader = loadersMap.get(extension.getName());
+    URLClassLoader loader = LOADERS_MAP.get(extension);
     if (loader == null) {
-      final Collection<File> jars = FileUtils.listFiles(extension, new String[]{"jar"}, false);
```

Comment:
@drcrallen I tried a hadoop batch ingestion job with this patch applied, and the task didn't pick up my hadoop-client jars.

Passing in the directory:
```
new URL[]{extension.toURL()}
```

doesn't seem to get the jars picked up, I had to change the URLClassLoader creation back to the original implementation with a url for each file to get the task to proceed.